### ISSUE TITLE
Contents of validator defs. file is deleted

### ISSUE THREAD

**Issue by MaxIsMyName** [sigp/lighthouse#2159]
## Description

Having run out of space on my Geth ethereum 1 node, I switched over to infura and deleted the chaindata directory and stopped the Geth service. After syncing back up with the beacon, which is now running without errors, the validator client is refusing to validate.

In the logs, all I can see is:

```
Jan 17 12:04:30 xxx lighthouse[1730]: Jan 17 12:04:30.412 CRIT Failed to start validator client        reason: Unable to open or create validator definitions: UnableToParseFile(EndOfStream)
Jan 17 12:04:30 xxx lighthouse[1730]: Jan 17 12:04:30.412 INFO Internal shutdown received              reason: Failed to start validator client
```

## Version

1.0.6

## Present Behaviour

Validator exits

## Expected Behaviour

Validator starts and validates on the network

**Comment by michaelsproul** [comment]
As discussed on Discord, I think this is a bug, as the validator definitions YAML ended up empty

I think a race condition like this is possible:


1. Lighthouse truncates file upon opening it in write mode (need to verify where this happens)
2. Disk is filled independently
3. Lighthouse fails to write the YAML because the disk is now full

**Comment by MaxIsMyName** [comment]
Solution that worked for me:

Re-import validator keys, something like this depending on your setup:
```
$ lighthouse --network mainnet account validator import --directory $HOME/eth2deposit-cli/validator_keys --datadir /var/lib/lighthouse
```

But note that you will have to edit the outputted validator_definitions.yml to add your password (https://lighthouse-book.sigmaprime.io/validator-management.html).

**Comment by paulhauner** [comment]
I believe I've seen this before as well. I'm going to flag it as a bug.

**Comment by michaelsproul** [comment]
This just happened to another user on Discord. I'll spend a few hours on Monday to get it fixed.

**Comment by michaelsproul** [comment]
Closed by #2338
